### PR TITLE
408 add `SplitChannel` transform for model outputs

### LINKED ISSUE #408: add SplitChannel transform for model output
**Is your feature request related to a problem? Please describe.**
`SplitChannel` transform can be used to split a multiple class label or model outputs.
For example, split multiple classes and compute the `mean_dice` metric separately.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #408 .

### Description
This PR implemented the `SplitChannel` post transform for model outputs.
Users can use it to split the outputs and labels then compute metrics separately.

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/transforms/post/dictionary.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            keys (hashable items): keys of the corresponding items to be transformed.
+                See also: :py:class:`monai.transforms.compose.MapTransform`
+            output_postfixes (list, tuple): the postfixes to construct keys to store splitted data.
+                for example: if the key of input data is `pred` and split 2 classes, the output
+                data keys will be: pred_(output_postfixes[0]), pred_(output_postfixes[1])
+            to_onehot (bool or list of bool): whether to convert the data to One-Hot format, default is False.
+        num_classes (int or list of int): the class number used to convert to One-Hot format if `to_onehot` is True.
+        """
+        super().__init__(keys)
+        assert isinstance(output_postfixes, (list, tuple)), 'must specify key postfixes to store splitted data.'
```
Comment (ericspod) [inline monai/transforms/post/dictionary.py:41]: Here we should be checking the condition and raising a ValueError instead of using assert.

Comment (Nic-Ma) [reply monai/transforms/post/dictionary.py]: updated in the latest patch.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Thanks very much for your review.
I updated the PR according to your comments.
And I got 2 small questions here:
1. Why you think "assert" is not good for parameters? not easy to "try-catch" outside?
2. Do you think it's OK to put post transforms at `/transforms/post/` or move to some other places like `/networks/`?

Thanks.

**Comment by ericspod** [comment]
Hi @Nic-Ma,
The thing with assert is that it is intended to be an internal sanity check rather than a mechanism to easily check some condition. Asserts can be turned off at runtime through a command line argument for efficiency, but we don't ever want input validation to be disabled. The assert statement is more convenient than having a `if ...: raise ...` kind of construct everywhere but it is better practice. Even better still would be some sort of design-by-contract facility so that we could define preconditions for inputs which are automatically turned into these checks for us. Something like [PyContracts](http://andreacensi.github.io/contracts/) or [icontract](https://github.com/Parquery/icontract) though I'm not recommending we start using these in MONAI without discussing it further. 

I think it's good to keep all the transforms we define in one place so yes under the `/transforms/post` directory. My feeling is that `/networks` should be strictly for the network definitions themselves, the layers they use, and a few utilities. Transforms are ancillary to the networks themselves even if defined in Pytorch.

**Pull request merged**